numpy/core/function_base.py

```diff
@@ -206,3 +206,76 @@ def logspace(start, stop, num=50, endpoint=True, base=10.0, dtype=None):
     if dtype is None:
         return _nx.power(base, y)
     return _nx.power(base, y).astype(dtype)
+
+
+def geomspace(start, stop, num=50, endpoint=True, dtype=None):
+    """
+    Return numbers spaced evenly on a log scale (a geometric progression).
+
+    This is similar to `logspace`, but with endpoints specified directly.
+    Each output sample is a constant multiple of the previous.
+
+    Parameters
+    ----------
+    start : float
+        The starting value of the sequence.
+    stop : float
+        The final value of the sequence, unless `endpoint` is False.
+        In that case, ``num + 1`` values are spaced over the
+        interval in log-space, of which all but the last (a sequence of
+        length `num`) are returned.
+    num : integer, optional
+        Number of samples to generate.  Default is 50.
+    endpoint : boolean, optional
+        If true, `stop` is the last sample. Otherwise, it is not included.
+        Default is True.
+    dtype : dtype
+        The type of the output array.  If `dtype` is not given, infer the data
+        type from the other input arguments.
+
+    Returns
+    -------
+    samples : ndarray
+        `num` samples, equally spaced on a log scale.
+
+    See Also
+    --------
+    logspace : Similar to geomspace, but with endpoints specified using log
+               and base.
+    linspace : Similar to geomspace, but with the samples uniformly distributed
+               in linear space, instead of log space.
+    arange : Similar to linspace, with the step size specified instead of the
+             number of samples. Note that, when used with a float endpoint, the
+             endpoint may or may not be included.
+
+    Examples
+    --------
+    >>> np.geomspace(10, 1000, num=4)
+    array([  100.        ,   215.443469  ,   464.15888336,  1000.        ])
+    >>> np.geomspace(100, 1000, num=3, endpoint=False)
+    array([ 100.        ,  215.443469  ,  464.15888336])
+    >>> np.geomspace(100, 1000, num=4, endpoint=False)
+    array([ 100.        ,  177.827941  ,  316.22776602,  562.34132519])
+    >>> np.geomspace(1, 256, num=9)
+    array([   1.,    2.,    4.,    8.,   16.,   32.,   64.,  128.,  256.])
+
+    Graphical illustration:
+
+    >>> import matplotlib.pyplot as plt
+    >>> N = 10
+    >>> x1 = np.geomspace(1, 10, N, endpoint=True)
+    >>> x2 = np.geomspace(1, 10, N, endpoint=False)
+    >>> y = np.zeros(N)
+    >>> plt.plot(x1, y, 'o')
+    [<matplotlib.lines.Line2D object at 0x...>]
+    >>> plt.plot(x2, y + 0.5, 'o')
+    [<matplotlib.lines.Line2D object at 0x...>]
+    >>> plt.ylim([-0.5, 1])
+    (-0.5, 1)
+    >>> plt.show()
+
+    """
+    if start == 0:
+        raise ValueError('Geometric sequence starting point must be nonzero')
+    return logspace(_nx.log10(start), _nx.log10(stop), num=num,
```

Comment:
Just out of curiosity, what motivated you to pick base 10 over say base e? I am not disagreeing, just genuinely curious.